stackslib/src/chainstate/burn/db/sortdb.rs

```diff
@@ -3365,12 +3370,14 @@ impl SortitionDB {
             info!("No migrator implementation given; `preprocessed_reward_sets` will not be prepopulated");
         }
 
-        let tx = self.tx_begin()?;
-        tx.execute(
-            "INSERT OR REPLACE INTO db_config (version) VALUES (?1)",
-            &["8"],
-        )?;
-        tx.commit()?;
+        if schema_version < 8 {
+            let tx = self.tx_begin()?;
+            tx.execute(
+                "INSERT OR REPLACE INTO db_config (version) VALUES (?1)",
+                &["8"],
+            )?;
+            tx.commit()?;
+        }
```

Comment:
Can you help me understand why this and the above change to `apply_schema_8_migration()` is necessary?

Reply:
This is because of where `apply_schema_8_migration` is called -- it is called after `apply_schema_9`. Could we instead move `apply_schema_9` (and later migrations) later or is there some reason that we cannot do that?

Reply:
I wasn't quite sure about the ordering of the migrations - it does make sense to execute 8 before 9, but i somewhat worry about unintended consequences, especially since these aren't just dropping/adding tables but doing other things as well.

Reply:
This is a bug introduced by e721bd5309.  The call to `apply_schema_9()` should happen _after_ the call to `apply_schema_8_migration()` in `SortitionDB::instantiate()`.  This is consistent with the behavior of `SortitionDB::check_schema_version_and_update()`.

Reply:
I've updated this to have schema 8's migration happen before 9 - mind giving this a review and resolving if it's all good?